providers/neo4j/src/test/java/net/javacrumbs/shedlock/provider/neo4j/Neo4jLockProviderIntegrationTest.java

```diff
@@ -0,0 +1,234 @@
+    void shouldNotUpdateOnInsertIfPreviousDidNotEndWhenUsingDbTime() {
+        shouldNotUpdateOnInsertIfPreviousDidNotEnd(true);
+    }
+
+    private void shouldNotUpdateOnInsertIfPreviousDidNotEnd(boolean usingDbTime) {
+        Neo4jStorageAccessor accessor = getAccessor(usingDbTime);
+
+        assertThat(
+            accessor.insertRecord(lockConfig(MY_LOCK, Duration.ofSeconds(10)))
+        ).isEqualTo(true);
+
+        Instant originalLockValidity = testUtils.getLockedUntil(MY_LOCK);
+
+        assertThat(
+            accessor.insertRecord(lockConfig(MY_LOCK, Duration.ofSeconds(10)))
+        ).isEqualTo(false);
+
+        assertThat(testUtils.getLockedUntil(MY_LOCK)).isEqualTo(originalLockValidity);
+    }
+
+    @Test
+    void shouldNotUpdateOtherLockConfigurationsWhenNotUsingDbTime() throws InterruptedException {
+        shouldNotUpdateOtherLockConfigurations(false);
+    }
+
+    @Test
+    void shouldNotUpdateOtherLockConfigurationsWhenUsingDbTime() throws InterruptedException {
+        shouldNotUpdateOtherLockConfigurations(true);
+    }
+
+    private void shouldNotUpdateOtherLockConfigurations(boolean usingDbTime) throws InterruptedException {
+        Neo4jStorageAccessor accessor = getAccessor(usingDbTime);
+
+        Duration lockAtMostFor = Duration.ofMillis(10);
+        assertThat(accessor.insertRecord(lockConfig(MY_LOCK, lockAtMostFor))).isEqualTo(true);
+        assertThat(accessor.insertRecord(lockConfig(OTHER_LOCK, lockAtMostFor))).isEqualTo(true);
+
+        Instant myLockLockedUntil = testUtils.getLockedUntil(MY_LOCK);
+        Instant otherLockLockedUntil = testUtils.getLockedUntil(OTHER_LOCK);
+
+        // wait for a while so there will be a difference in the timestamp
+        // when system time is used seems there is no milliseconds in the timestamp so to make a difference we have to wait for at least a second
+        sleep(1000);
+
+
+        // act
+        assertThat(accessor.updateRecord(new LockConfiguration(now(), MY_LOCK, lockAtMostFor, Duration.ZERO))).isEqualTo(true);
+
+
+        // assert
+        assertThat(testUtils.getLockedUntil(MY_LOCK)).isAfter(myLockLockedUntil);
+        // check that the other lock has not been affected by "my-lock" update
+        assertThat(testUtils.getLockedUntil(OTHER_LOCK)).isEqualTo(otherLockLockedUntil);
+    }
+
+    @NonNull
+    private LockConfiguration lockConfig(String myLock, Duration lockAtMostFor) {
+        return new LockConfiguration(now(), myLock, lockAtMostFor, Duration.ZERO);
+    }
+
+    @NonNull
+    protected Neo4jStorageAccessor getAccessor(boolean usingDbTime) {
+        return new Neo4jStorageAccessor(this.testUtils.getDriver(), "shedlock");
+    }
+
+
+    @Test
+    @Disabled
+    public void shouldNotFailIfKeyNameTooLong() {
+        LockConfiguration configuration = lockConfig("lock name that is too long Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.");
```

Comment:
Why is this disabled?

Reply:
Initially it was disabled in JDBC module which I used as copy source. But a test like this does not make sense as the limit is machine dependent and equal to the limit of Java String. I removed it.